Translate hello-world.md (#38)
reactjs/fa.reactjs.org: content/docs/hello-world.md

---
id: hello-world
title: سلام دنیا
permalink: docs/hello-world.html
prev: cdn-links.html
next: introducing-jsx.html
---

> کوچکترین مثال ری‌اکت به شکل زیر است:

Reviewer: ```suggestion
کوچکترین مثال ری‌اکت شبیه این است:
```

Reviewer: <p dir='rtl' align='right'>
کوچکترین مثال ری‌اکت شبیه این است:
</p>

```js
ReactDOM.render(
  <h1>Hello, world!</h1>,
  document.getElementById('root')
);
```

> یک عنوان را روی صفحه نمایش می‌دهد که می‌گوید: "Hello, world!"

Reviewer: ```suggestion
این مثال یک عنوان را روی صفحه نمایش می‌دهد که می‌گوید: "Hello, world!"
```

Reviewer: <p dir='rtl' align='right'>
این مثال یک عنوان را روی صفحه نمایش می‌دهد که می‌گوید:
</p>

**[روی CodePen امتحان کنید](codepen://hello-world)**

> روی لینک بالا کلیک کنید تا یک ویرایشگر آنلاین باز شود. هر تغییری که می‌خواهید اعمال کنید و تاثیر آن را بر خروجی مشاهده کنید. بیشتر صفحات این راهنما، مثال‌های قابل ویرایشی مانند بالا دارند.

Reviewer: ```suggestion
روی لینک بالا کلیک کنید تا یک ویرایشگر آنلاین باز شود. هر تغییری که می‌خواهید اعمال کنید و تاثیر آن را بر خروجی مشاهده کنید. بیشتر صفحات در این راهنما، مثال‌های قابل ویرایشی مانند بالا دارند.
```

## چگونه این راهنما را بخوانیم {#how-to-read-this-guide}

> در این راهنما، ما به بررسی اجزا سازنده برنامه‌های ری‌اکت یعنی: المنت‌ها و کامپوننت‌ها می‌پردازیم. زمانی که شما به آن‌ها مسلط شدید، می‌توانید برنامه‌های پیچیده را از تکه‌های کوچکتر با قابلیت استفاده محدد بسازید.

Reviewer 2: `مجدد`

> در این راهنما، ما به بررسی اجزا سازنده برنامه‌های ری‌اکت یعنی: المنت‌ها و کامپوننت‌ها می‌پردازیم. زمانی که شما به آن‌ها مسلط شدید، می‌توانید برنامه‌های پیچیده را از تکه‌های کوچکتر با قابلیت استفاده مجدد بسازید.

Reviewer: ```suggestion
در این راهنما، ما به بررسی اجزا سازنده برنامه‌های ری‌اکت یعنی: المنت‌ها و کامپوننت‌ها می‌پردازیم. زمانی که شما به آن‌ها مسلط شوید، می‌توانید برنامه‌های پیچیده را از تکه‌های کوچکتر با قابلیت استفاده مجدد بسازید.
```

>راهنما
>

> >این راهنما برای افرادی طراحی شده‌است که ترجیح می‌دهند **مفاهیم را گام‌به‌گام بیاموزند**. اگر شما ترجیح می‌دهید انجام دهید و یاد بگیرید، به [آموزش عملی](/tutorial/tutorial.html) ما نگاهی بیاندازید. شاید این راهنما و آن آموزش برای شما مکمل یکدیگر باشند.

Reviewer: ```suggestion
>این راهنما برای افرادی طراحی شده‌است که ترجیح می‌دهند **مفاهیم را گام‌به‌گام بیاموزند**. اگر شما ترجیح می‌دهید به وسیله انجام‌دادن یاد بگیرید، به [آموزش عملی](/tutorial/tutorial.html) ما نگاهی بیاندازید. شاید این راهنما و آن آموزش برای شما مکمل یکدیگر باشند.
```

> این اولین فصل از راهنمای گام‌به‌گام درباره مفاهیم اصلی ری‌اکت است. شما می‌توانید لیستی از تمام سرفصل‌ها را در منوی سمت چپ مشاهده‌ کنید. اگر با موبایل درحال خواندن این راهنما هستید، با زدن دکمه پایین سمت چپ صفحه می‌توانید به منو دسترسی پیدا کنید.

Reviewer: ```suggestion
این اولین فصل از راهنمای گام‌به‌گام درباره مفاهیم اصلی ری‌اکت است. شما می‌توانید لیستی از تمام سرفصل‌ها را در نوار کناری مشاهده‌ کنید. اگر با موبایل درحال خواندن این راهنما هستید، با زدن دکمه در گوشه پایین سمت چپ صفحه می‌توانید به منو دسترسی پیدا کنید.
```

> دانش هر فصل از این راهنما براساس مفاهیم ارائه‌شده در فصل‌های پیشین بنا می‌شود. **شما می‌توانید بیشتر ری‌اکت را با مطالعه سرفصل‌های راهنمای "مفاهیم پایه‌ای" که در منوی سمت چپ آورده‌ شده‌است، یاد بگیرید.** برای مثال، [“مقدمه‌ای بر JSX”](/docs/introducing-jsx.html) فصل بعدی است.

Reviewer: ```suggestion
هر فصل در این راهنما براساس دانش معرفی‌شده در فصل‌های پیشین بنا می‌شود. **شما می‌توانید بیشتر ری‌اکت را با مطالعه فصل‌های راهنمای "مفاهیم اصلی" به ترتیبی که در نوار کناری آورده‌‌ شده است، یاد بگیرید.** برای مثال، [“مقدمه‌ای بر JSX”](/docs/introducing-jsx.html) فصل بعدی بعد از این فصل است.
```

Reviewer: <p dir='rtl' align='right'>
هر فصل در این راهنما براساس دانش معرفی‌شده در فصل‌های پیشین بنا می‌شود. **شما می‌توانید بیشتر ری‌اکت را با مطالعه فصل‌های راهنمای "مفاهیم اصلی" به ترتیبی که در نوار کناری آورده‌‌ شده است، یاد بگیرید.** برای مثال، [“مقدمه‌ای بر JSX”](/docs/introducing-jsx.html) فصل بعدی بعد از این فصل است.
</p>

> ## تخمین سطح دانش {#knowledge-level-assumptions}

Reviewer: ```suggestion
## فرض‌های سطح دانش {#knowledge-level-assumptions}
```

> ری‌اکت یک کتابخانه جاوااسکریپت است، و بنابراین ما بر این گمانیم که شما  درک پایه‌ای از زبان جاوااسکریپت دارید. **اگر خیلی مطمئن نیستید، ما پیشنهاد می‌کنیم به [آموزش عمیق جاوااسکریپت](https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-introduction_to_JavaScript) مراجعه کنید تا سطح دانش خود را بررسی کنید** و همچنین این راهنما را به آسانی دنبال کنید. این کار ممکن است برای شما بین ۳۰ دقیقه تا یک ساعت به طول بیانجامد، اما در نتیجه این حس را نخواهید داشت که [با خواندن این راهنما] ری‌اکت و جاوااسکریپت را همزمان یاد می‌گیرید.

Reviewer: ```suggestion
ری‌اکت یک کتابخانه جاوااسکریپت است، و بنابراین ما فرض می‌کنیم که شما درک پایه‌ای از زبان جاوااسکریپت دارید. **اگر خیلی مطمئن نیستید، پیشنهاد می‌کنیم [یک آموزش جاوااسکریپت را طی کنید](https://developer.mozilla.org/en-US/docs/Web/JavaScript/A_re-introduction_to_JavaScript) تا سطح دانش خود را بررسی کنید** و بتوانید این راهنما را بدون گمراهی دنبال کنید. این کار ممکن است برای شما بین ۳۰ دقیقه تا یک ساعت به طول بیانجامد، اما در نتیجه این حس را نخواهید داشت که [با خواندن این راهنما] ری‌اکت و جاوااسکریپت را همزمان یاد می‌گیرید.
```

>نکته
>

> >این راهنما در اغلب مثال‌ها از نگارش جدید جاوااسکریپت استفاده می‌کند. اگر در چند سال اخیر با جاوااسکریپت کار نکرده‌اید، این [این سه مورد](https://gist.github.com/gaearon/683e676101005de0add59e8bb345340c) بیشتر موارد را به شما یاد می‌دهد.

Reviewer: ```suggestion
>این راهنما در اغلب مثال‌ها از نگارش جدید جاوااسکریپت استفاده می‌کند. اگر در چند سال اخیر با جاوااسکریپت کار نکرده‌اید، [این سه مورد](https://gist.github.com/gaearon/683e676101005de0add59e8bb345340c) بیشتر موارد را به شما یاد می‌دهد.
```

## بیایید شروع کنیم! {#lets-get-started}

> به پایین اسکرول کنید و لینک [فصل بعدی این راهنما](/docs/introducing-jsx.html) را پیش از footer وبسایت پیدا می‌کنید.

Reviewer: ```suggestion
به پایین اسکرول کردن ادامه دهید و لینک [فصل بعدی این راهنما](/docs/introducing-jsx.html) را پیش از footer وبسایت پیدا می‌کنید.
```
